vmngclient/session.py

```diff
@@ -388,16 +393,25 @@ def session_type(self) -> SessionType:
         return self._session_type
 
     @property
-    def platform_version(self) -> Optional[Version]:
+    def platform_version(self) -> str:
         return self._platform_version
 
     @platform_version.setter
-    def platform_version(self, version: Version):
+    def platform_version(self, version: str):
         self._platform_version = version
-        self._api_version = Version(f"{version.major}.{version.minor}")
+
+        try:
+            parsed_version = parse_version(version)
+            self._api_version = Version(f"{parsed_version.major}.{parsed_version.minor}")
+        except InvalidVersion:
+            try:
+                parsed_version = parse_version(".".join(version.split(".")[:2]))
+                self._api_version = Version(f"{parsed_version.major}.{parsed_version.minor}")
+            except InvalidVersion:
+                self.logger.warning(f"vManage-client was not able to recognize API version from: {version}")
 
     @property
-    def api_version(self) -> Version:
+    def api_version(self) -> Optional[Version]:
```

Comment:
@tehAgitto what if we cloud a little bit simplify and keep previous signature ? We will do it by using semi Null object DP on Version. It'll always exist and properties will present a state of it